packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -23,12 +25,17 @@ import { CloudFormationStack } from './util/cloudformation';
 export async function tryHotswapDeployment(
   sdkProvider: SdkProvider, assetParams: { [key: string]: string },
   cloudFormationStack: CloudFormationStack, stackArtifact: cxapi.CloudFormationStackArtifact,
+  hotswapLogMonitor?: CloudWatchLogEventMonitor,
 ): Promise<DeployStackResult | undefined> {
   // resolve the environment, so we can substitute things like AWS::Region in CFN expressions
   const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
   // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
   // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
   const sdk = await sdkProvider.forEnvironment(resolvedEnv, Mode.ForWriting);
+
+  // add the sdk to the hotswapLogMonitor so that it also has the
+  // correct credentials to monitor CloudWatch log groups
+  hotswapLogMonitor?.addSDK(sdk);
```

Comment:
We write acronyms in camelcase:

```suggestion
  hotswapLogMonitor?.addSdk(sdk);
```